File: Examples/DynamicFieldsExample/Classes/ISO3166CountryValueTransformer.m
//
//  ISO3166CountryValueTransformer.m
//  DynamicFieldsExample
//
//  Created by Bart Vandendriessche on 17/03/14.
//  Copyright (c) 2014 Charcoal Design. All rights reserved.
//

#import "ISO3166CountryValueTransformer.h"

@implementation ISO3166CountryValueTransformer

+ (Class)transformedValueClass
{
    return [NSString class];
}

+ (BOOL)allowsReverseTransformation
{
    return NO;
}

- (id)transformedValue:(id)value
{
    return value? [[NSLocale currentLocale] displayNameForKey:NSLocaleCountryCode value:value]: nil;
}

@end

File: Examples/SwiftExample/SwiftExample/ISO3166CountryValueTransformer.swift
//
//  ISO3166CountryValueTransformer.swift
//  SwiftExample
//
//  Created by Nick Lockwood on 29/09/2014.
//  Copyright (c) 2014 Nick Lockwood. All rights reserved.
//

import UIKit

class ISO3166CountryValueTransformer: NSValueTransformer {
   
    override class func transformedValueClass() -> AnyClass {
        
        return NSString.self
    }
    
    override class func allowsReverseTransformation() -> Bool {
        
        return false
    }
    
    override func transformedValue(value: AnyObject?) -> AnyObject? {
        
        if value != nil {
            return NSLocale(localeIdentifier: "en_US").displayNameForKey(NSLocaleCountryCode, value:value!) ?? value
        }
        return nil
    }
}
